src/samples/mqtt/bluemix.c

```diff
@@ -0,0 +1,355 @@
+publish_callback(void *vdata)
+{
+    struct bluemix_data *data = (struct bluemix_data *)vdata;
+    struct sol_buffer payload = SOL_BUFFER_INIT_EMPTY;
+    struct sol_mqtt_message message;
+    char topic[] = "iot-2/evt/sensorData/fmt/json-iotf";
+    int r;
+    double v;
+    static int pulse = 0;
+
+    pulse = (pulse + 1) % 360;
+    v = sin(pulse * PI / 180) * 10;
+
+    r = sol_buffer_append_printf(&payload, SENSOR_FMT, v * 1.25, v * 1, v * 0.75, v * 20, v * 15, v * 10);
+    SOL_INT_CHECK(r, < 0, false);
+
+    message = (struct sol_mqtt_message){
+        .topic = topic,
+        .payload = &payload,
+        .qos = SOL_MQTT_QOS_EXACTLY_ONCE,
+        .retain = false,
+    };
+
+    r = sol_mqtt_publish(data->mqtt, &message);
+
+    sol_buffer_fini(&payload);
+
+    SOL_INT_CHECK(r, != 0, false);
+
+    return true;
+}
+
+static bool
+try_reconnect(void *data)
+{
+    return sol_mqtt_reconnect((struct sol_mqtt *)data) != 0;
+}
+
+static void
+on_connect(void *vdata, struct sol_mqtt *mqtt)
+{
+    struct bluemix_data *data = (struct bluemix_data *)vdata;
+
+    if (sol_mqtt_get_connection_status(mqtt) != SOL_MQTT_CONNECTED) {
+        SOL_WRN("Unable to connect, retrying...");
+        sol_timeout_add(1000, try_reconnect, mqtt);
+        return;
+    }
+
+    data->publish_timeout = sol_timeout_add(100, publish_callback, data);
+    if (!data->publish_timeout)
+        SOL_WRN("Unable to setup callback");
+}
+
+static void
+on_disconnect(void *data, struct sol_mqtt *mqtt)
+{
+    SOL_INF("Reconnecting...");
+    sol_timeout_add(1000, try_reconnect, mqtt);
+}
+
+static struct sol_mqtt_config config = {
+    .api_version = SOL_MQTT_CONFIG_API_VERSION,
+    .clean_session = true,
+    .keepalive = 60,
+    .username = "use-token-auth",
+    .handlers = {
+        .connect = on_connect,
+        .disconnect = on_disconnect,
+    },
+};
+
+static bool
+mqtt_init(void *vdata)
+{
+    struct bluemix_data *data = (struct bluemix_data *)vdata;
+    struct sol_buffer buffer = SOL_BUFFER_INIT_EMPTY;
+    int r;
+
+    SOL_NULL_CHECK_GOTO(data, error);
+
+    r = sol_buffer_append_printf(&buffer, "%s.messaging.internetofthings.ibmcloud.com", data->org_id);
+    SOL_INT_CHECK_GOTO(r, < 0, error);
+
+    config.client_id = data->client_id;
+    config.password = data->auth_token;
+
+    data->mqtt = sol_mqtt_connect(buffer.data, 1883, &config, data);
+
+    sol_buffer_fini(&buffer);
+
+    if (!data->mqtt) {
+        SOL_WRN("Unable to create MQTT session. Retrying...");
+        return true;
+    }
+
+    SOL_INF("Sending sensor data");
+
+    return false;
+
+error:
+    SOL_WRN("Unable to initialize MQTT");
+    sol_quit();
+
+    return false;
+}
+
+/**
+ * Authenticate to HTTP Server
+ */
+
+static bool
+json_token_to_string(struct sol_json_token *token, char **out)
+{
+    if (sol_json_token_get_type(token) != SOL_JSON_TYPE_STRING)
+        return false;
+    free(*out);
+    *out = strndup(token->start + 1, token->end - token->start - 2);
```

Comment:
so far, so good. But we're adding API to unescape the string, this will be the recommended way to extract the string. Sync with @otaviobp so either you or him update this function to use it.